lightkurve/injection.py

```diff
@@ -0,0 +1,288 @@
+                                         self.var)
+
+    def sample(self):
+        """Chooses values from Gaussian distribution.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        values : array-like
+            Returns array of randomly chosen values.
+        """
+        return np.random.normal(self.mean, self.var)
+
+    def plot(self):
+        """Plots specified distribution."""
+        t = np.linspace(self.mean - 3*self.var, self.mean + 3*self.var, 100)
+        vals = norm.pdf(t, self.mean, self.var)
+        plt.plot(t, vals)
+        plt.ylim(0, np.max(vals))
+
+
+class TransitModel(object):
+    """
+    Implements a class for creating a transiting model using ktransit.
+    When you initialize the model, you must set parameters for the star.
+
+    Attributes
+    ----------
+    zpt : float
+        A photometric zeropoint
+    **kwargs : dict
+        Star parameters
+
+    """
+
+    def __init__(self):
+        import ktransit
+        self.multiplicative = True
+        self.model = ktransit.LCModel()
+
+    def __repr__(self):
+        return 'TransitModel(' + str(self.__dict__) + ')'
+
+    def add_star(self, zpt=1.0, **kwargs):
+        """Initializes the star.
+
+        Parameters
+        ----------
+        Default values are those initialized in TransitModel.
+        A parameter must be defined in the initialization of
+        TransitModel if it is to be changed in add_planet.
+
+        zpt : float
+            A photometric zeropoint
+        **kwargs : dict
+            Dictonary of planet parameters. Options are:
+                rho : stellar density
+                ld1, ld2, ld3, ld3 : limb darkening coefficients
+                dil : transit dilution fraction
+                veloffset : (not sure what this is)
+        """
+
+        self.zpt = zpt
+
+        self.star_params = {}
+        for key, value in kwargs.items():
+            if isinstance(value, (GaussianDistribution, UniformDistribution)):
+                self.star_params[key] = value.sample()
+            else:
+                self.star_params[key] = value
+
+        self.model.add_star(zpt=self.zpt, **self.star_params)
+
+    def add_planet(self, period, rprs, T0, **kwargs):
+        """Adds a planet to TransitModel object.
+
+        Parameters
+        ----------
+        period : float
+            Orbital period of new planet
+        rprs : float
+            Planet radius/star radius of new planet
+        T0 : float
+            A transit mid-time
+        **kwargs : dict
+            Dictonary of planet parameters. Options are:
+                impact: an impact parameter
+                ecosw, esinw : an eccentricity vector
+                occ : a secondary eclipse depth
+                rvamp : (not sure)
+                ell : (not sure)
+                alb : (not sure)
+        """
+
+        if isinstance(period, (GaussianDistribution, UniformDistribution)):
```

Comment:
Rather than having 3 if statements, could we make this into a loop over all the attributes? Something you have below?